### DISCUSSION TITLE
Optimizing DDS to PCIe Data Transfer

### DISCUSSION THREAD

**Discussion by aqnaazeez** [eProsima/Fast-DDS#4672] [Q&A]
The data send through the DDS layer is send to UDP through a buffer, we have replaced the UDP layer with PCIE, which has a buffer. Is there any method to copy the data from DDS layer directly to the PCIe buffer in kernel space. Currently we’re copying the DDS buffer data to a PCIE buffer.

**Comment by elianalf** [comment]
Hi @aqnaazeez, 
you might find interesting this other [discussion](https://github.com/eProsima/Fast-DDS/discussions/4623). If you have already read this post or your project is further along and you want more specific information, at this moment there isn't a method to copy the data from DDS layer directly to the PCIe buffer in kernel space so that would probably imply a custom solution. If you need more help in relation to that, do not hesitate contacting our commercial support: [support@eprosima.com](mailto:support@eprosima.com)
